add.go

```diff
@@ -161,6 +168,24 @@ func (b *Builder) Add(destination string, extract bool, source ...string) error
 				if err := copyFileWithTar(gsrc, d); err != nil {
 					return errors.Wrapf(err, "error copying %q to %q", gsrc, d)
 				}
+
+				// Set permissions. For now both uid and gid need to be set and they have to be numbers.
+				if options.Chown[0] != "" && options.Chown[1] != "" {
+					// TODO: figure out uid/id if user/group names was passed in instead of the real uid/gid.
```

Comment:
I believe there is code for this already written to handle passing of of USER string.